[Example] A version conflict for spring-core causes example startup to fail

## Bug Report

### Which version of ShardingSphere did you use?
sharding-spring-boot-mybatis-example, from master branch

### Expected behavior
execution succeed

### Actual behavior
failed to startup
```
Exception in thread "main" java.lang.IllegalArgumentException: Cannot instantiate interface org.springframework.context.ApplicationContextInitializer : org.springframework.boot.context.ConfigurationWarningsApplicationContextInitializer
	at org.springframework.boot.SpringApplication.createSpringFactoriesInstances(SpringApplication.java:439)
	at org.springframework.boot.SpringApplication.getSpringFactoriesInstances(SpringApplication.java:418)
	at org.springframework.boot.SpringApplication.getSpringFactoriesInstances(SpringApplication.java:409)
	at org.springframework.boot.SpringApplication.<init>(SpringApplication.java:259)
	at org.springframework.boot.SpringApplication.<init>(SpringApplication.java:240)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1242)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1230)
	at org.apache.shardingsphere.example.sharding.spring.boot.mybatis.ShardingSpringBootMybatisExample.main(ShardingSpringBootMybatisExample.java:37)
Caused by: java.lang.NoSuchMethodError: org.springframework.core.KotlinDetector.isKotlinReflectPresent()Z
	at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:191)
	at org.springframework.boot.SpringApplication.createSpringFactoriesInstances(SpringApplication.java:435)
	... 7 more
```

### Reason analyze (If you can)
In PR #14883, the dependency on `spring-framework`  of example was adjusted:
````
# from
<spring-framework.version>5.0.13.RELEASE</spring-framework.version>
<spring-boot.version>2.0.9.RELEASE</spring-boot.version>

# to
<spring-framework.version>5.2.19.RELEASE</spring-framework.version>
<spring-boot.version>2.0.9.RELEASE</spring-boot.version>
````
Since spring-boot 2.0.9.RELEASE depends on Spring Framework 5.0.13.RELEASE by default[1], this causes the example to introduce both `spring-core 5.0 0.13.RELEASE` and `spring-core 5.2.19.RELEASE`.
<img width="638" alt="image" src="https://user-images.githubusercontent.com/5668787/153812590-b1fd09bf-42d3-4707-8ec6-37cde714ef72.png">
Meanwhile, `KotlinDetector` in 5.0.13 does not have a method called `isKotlinReflectPresent`, it exists in 5.2.19.


### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.
run `ShardingSpringBootMybatisExample`


### Reference
1: https://github.com/spring-projects/spring-boot/releases/tag/v2.0.9.RELEASE

--- a/examples/pom.xml
+++ b/examples/pom.xml
@@ -42,8 +42,8 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.build.locale>zh_CN</project.build.locale>
         <aspectjweaver.version>1.8.9</aspectjweaver.version>
-        <spring-framework.version>5.2.19.RELEASE</spring-framework.version>
-        <spring-boot.version>2.0.9.RELEASE</spring-boot.version>
+        <spring-framework.version>5.2.15.RELEASE</spring-framework.version>
+        <spring-boot.version>2.3.12.RELEASE</spring-boot.version>
         <hikari-cp.version>3.4.2</hikari-cp.version>
         <mysql-connector-java.version>5.1.47</mysql-connector-java.version>
         <postgresql.version>42.2.5</postgresql.version>
@@ -51,8 +51,9 @@
         <slf4j.version>1.7.7</slf4j.version>
         <logback.version>1.2.0</logback.version>
         <lombok.version>1.18.20</lombok.version>
-        <mybatis.version>3.5.1</mybatis.version>
-        <mybatis-spring.version>2.0.1</mybatis-spring.version>
+        <mybatis.version>3.5.5</mybatis.version>
+        <mybatis-spring.version>2.0.5</mybatis-spring.version>
+        <mybatis-spring-boot.version>2.1.3</mybatis-spring-boot.version>
 
         <jpa.version>1.0.0.Final</jpa.version>
         <hibernate.version>5.2.18.Final</hibernate.version>
@@ -260,7 +261,7 @@
             <dependency>
                 <groupId>org.mybatis.spring.boot</groupId>
                 <artifactId>mybatis-spring-boot-starter</artifactId>
-                <version>${mybatis-spring.version}</version>
+                <version>${mybatis-spring-boot.version}</version>
             </dependency>
             <dependency>
                 <groupId>org.codehaus.btm</groupId>

